tests/weekday.rs

```diff
@@ -0,0 +1,259 @@
+#[cfg(feature = "std")]
+extern crate core;
+
+use core::str::FromStr;
+
+use hifitime::{Duration, Epoch, TimeUnits, Unit, Weekday};
+
+#[test]
+fn test_basic_ops() {
+    assert_eq!(Weekday::default(), Weekday::Monday);
+
+    let monday = Weekday::default();
+
+    assert_eq!(monday - 1, Weekday::Sunday);
+    assert_eq!(monday - 2, Weekday::Saturday);
+    assert_eq!(monday - 3, Weekday::Friday);
+    assert_eq!(monday - 4, Weekday::Thursday);
+    assert_eq!(monday - 5, Weekday::Wednesday);
+    assert_eq!(monday - 6, Weekday::Tuesday);
+    assert_eq!(monday - 7, monday);
+    assert_eq!(monday - 8, Weekday::Sunday);
+    assert_eq!(monday - 9, Weekday::Saturday);
+    assert_eq!(monday - 13, Weekday::Tuesday);
+    assert_eq!(monday - 14, monday);
+    assert_eq!(monday - 15, Weekday::Sunday);
+
+    let i: i8 = -1;
+    let weekday: Weekday = i.into();
+    assert_eq!(weekday, Weekday::Sunday);
+    let i: i8 = -2;
+    let weekday: Weekday = i.into();
+    assert_eq!(weekday, Weekday::Saturday);
+}
+
+#[test]
+fn test_weekday_differences() {
+    let monday = Weekday::Monday;
+
+    for day_num in 0..15_u8 {
+        let day = Weekday::from(day_num);
+        let neg_delta: Duration = monday - day;
+        let pos_delta: Duration = day - monday;
+        // Check reciprocity
+        if day_num % 7 == 0 {
+            assert_eq!(pos_delta + neg_delta, Duration::ZERO);
+        } else {
+            assert_eq!(pos_delta + neg_delta, 7 * Unit::Day);
+        }
+        // Check actual value
+        assert_eq!(neg_delta, i64::from(day_num % 7) * Unit::Day);
+    }
+
+    // Start in the middle of the week
+    for day_num in 0..15_u8 {
+        let day = Weekday::from(day_num);
+        let neg_delta: Duration = Weekday::Wednesday - day;
+        let pos_delta: Duration = day - Weekday::Wednesday;
+        // Check reciprocity
+        if day_num % 7 == 2 {
+            assert_eq!(pos_delta + neg_delta, Duration::ZERO);
+        } else {
+            assert_eq!(pos_delta + neg_delta, 7 * Unit::Day);
+        }
+        // Check actual value
+        if day_num % 7 <= 2 {
+            assert_eq!(pos_delta, i64::from(2 - day_num % 7) * Unit::Day);
+        } else {
+            assert_eq!(neg_delta, i64::from(day_num % 7 - 2) * Unit::Day);
+        }
+        // Test FromStr
+        assert_eq!(Weekday::from_str(&format!("{day}")).unwrap(), day);
+    }
+}
+
+#[test]
+fn test_next() {
```

Comment:
interesting approach with `next` and `previous`

Reply:
I think it can be useful as a general use-case, for setting cookies or whatever.